activesupport/lib/active_support/encrypted_file.rb

```diff
@@ -75,6 +81,9 @@ def writing(contents)
 
       def encrypt(contents)
         encryptor.encrypt_and_sign contents
+      rescue ArgumentError => e
+        raise InvalidKeyLengthError if e.message.match?(/key must be \d+ bytes/)
```

Comment:
How about validating the key length upfront instead? That would avoid coupling with an [implementation detail](https://github.com/ruby/openssl/blob/3d160913bfce3e40fcc017524cb28f7507395713/ext/openssl/ossl_cipher.c#L481) of the openssl gem, and provide the same feedback when decrypting an existing file too.

Reply:
Yes, I debated this.  My thinking at the time was that rescuing was the simpler solution, and, since it was only a minor quality-of-life improvement, it would be tolerable if the OpenSSL gem somehow broke it between Rails releases.

I changed the code to follow your suggestion, but I limited it to encryption only because I feel like it would be inconsistent to not raise an `ActiveSupport::MessageEncryptor::InvalidMessage` when decrypting with an incorrect key.  And also because always raising when a key has invalid length would have complicated interaction with `raise_if_missing_key: false`.  We could skip the length check when the key is missing, but then encrypting would still raise a confusing message.